services/intersection-api/api/src/main/java/us/dot/its/jpo/ode/api/accessors/events/LaneDirectionOfTravelEvent/LaneDirectionOfTravelEventRepositoryImpl.java

```diff
@@ -0,0 +1,167 @@
+import org.springframework.data.mongodb.core.query.Criteria;
+import org.springframework.data.mongodb.core.query.Query;
+import org.springframework.stereotype.Component;
+
+import systems.uom.common.USCustomary;
+import tech.units.indriya.quantity.Quantities;
+import tech.units.indriya.unit.Units;
+import us.dot.its.jpo.conflictmonitor.monitor.models.events.LaneDirectionOfTravelEvent;
+import org.springframework.data.domain.Sort;
+
+import org.springframework.data.mongodb.core.aggregation.Aggregation;
+import org.springframework.data.mongodb.core.aggregation.AggregationResults;
+import org.springframework.data.mongodb.core.aggregation.ArithmeticOperators;
+import org.springframework.data.mongodb.core.aggregation.DateOperators;
+
+import us.dot.its.jpo.ode.api.ConflictMonitorApiProperties;
+import us.dot.its.jpo.ode.api.models.IDCount;
+
+@Component
+public class LaneDirectionOfTravelEventRepositoryImpl implements LaneDirectionOfTravelEventRepository {
+
+    @Autowired
+    private MongoTemplate mongoTemplate;
+
+    @Autowired
+    ConflictMonitorApiProperties props;
+
+    private final Quantity<Length> one_centimeter = Quantities.getQuantity(1, MetricPrefix.CENTI(Units.METRE));
+    private final Double ONE_CENTIMETER_IN_FEET = one_centimeter.to(USCustomary.FOOT).getValue().doubleValue();
+
+    private String collectionName = "CmLaneDirectionOfTravelEvent";
+
+    public Query getQuery(Integer intersectionID, Long startTime, Long endTime, boolean latest) {
+        Query query = new Query();
+
+        if (intersectionID != null) {
+            query.addCriteria(Criteria.where("intersectionID").is(intersectionID));
+        }
+        Date startTimeDate = new Date(0);
+        Date endTimeDate = new Date();
+
+        if (startTime != null) {
+            startTimeDate = new Date(startTime);
+        }
+        if (endTime != null) {
+            endTimeDate = new Date(endTime);
+        }
+
+        query.addCriteria(Criteria.where("eventGeneratedAt").gte(startTimeDate).lte(endTimeDate));
+        if (latest) {
+            query.with(Sort.by(Sort.Direction.DESC, "eventGeneratedAt"));
+            query.limit(1);
+        }else{
+            query.limit(props.getMaximumResponseSize());
+        }
+
+
+        return query;
+    }
+
+    public long getQueryResultCount(Query query) {
+        return mongoTemplate.count(query, LaneDirectionOfTravelEvent.class, collectionName);
+    }
+
+    public long getQueryFullCount(Query query){
+        int limit = query.getLimit();
+        query.limit(-1);
+        long count = mongoTemplate.count(query, LaneDirectionOfTravelEvent.class, collectionName);
+        query.limit(limit);
+        return count;
+    }
+
+    public List<LaneDirectionOfTravelEvent> find(Query query) {
+        return mongoTemplate.find(query, LaneDirectionOfTravelEvent.class, collectionName);
+    }
+
+    public List<IDCount> getLaneDirectionOfTravelEventsByDay(int intersectionID, Long startTime, Long endTime){
```

Comment:
suggestion(non-blocking): rename for clarity. "...ByDay" makes it seem like we're expecting a single day for query but this method can find objects across any given start/end times

Reply:
I just did a massive renaming of all of the "getBsmEventsByDay" similar methods to something like "getAggregatedDailyBsmEventCounts"